### ISSUE TITLE
Segmentation fault ./start.sh

### ISSUE THREAD

**Issue by teracube** [CortexPE/TeaSpoon#136]
## Issue Description

As i comment here #129
My server crash
I only update TeaSpoon_dev-81 to TeaSpoon_dev-83
no other modification

Server crash and console show this error "segmentation error"
bash : ligne 1 : 4318 Erreur de segmentation ./start.sh
or
bash : ligne 1 : 24807 Erreur de segmentation ./start.sh

no other error
no crashdump found

My tests 

1/delete teaspoon.phar and config then copy a new teaspoon.phar file then restart the server =) server crash after 3 to 10 minutes with 5 to 10 players

2/ delete ender/nether and restart server.  oknew ender /nether generated on restart but =) server crash after 3 to 10 minutes with 5 to 10 players

2/If i remove fully TeaSpoon plugin  server is OK, no problem but no TeaSpoon :(
:( :( :(

for information my server is multiwords there is 6 worlds on my server (4 + ender + nether)

## Steps to Reproduce

1. ... PocketMine-MP version [33m1.7dev-554[m "[REDACTED]" (API 3.0.0-ALPHA10)
2. ...TeaSpoon v1.0.0 version dev 83
3. ... 

Linux
PHP 7.2 RC4

## Crashdump, Backtrace or Other Files

```
-
]0;PocketMine-MP 1.7dev-561 | Online 12/50 | Memory 234.34/3.8/418.51/1269.02 MB @ 11 threads | U 147.79 D 5.45 kB/s | TPS 19.72 | Load 22.66%]0;PocketMine-MP 1.7dev-561 | Online 12/50 | Memory 251.04/3.95/424.38/1275.16 MB @ 11 threads | U 115.14 D 7.52 kB/s | TPS 20 | Load 21.15%]0;PocketMine-MP 1.7dev-561 | Online 12/50 | Memory 259.17/3.95/429.09/1281.16 MB @ 11 threads | U 155.88 D 8.09 kB/s | TPS 20 | Load 21.88%]0;PocketMine-MP 1.7dev-561 | Online 12/50 | Memory 259.2/3.95/429.09/1281.16 MB @ 11 threads | U 63.42 D 7.77 kB/s | TPS 20 | Load 16.12%]0;PocketMine-MP 1.7dev-561 | Online 12/50 | Memory 259.23/3.95/429.34/1281.16 MB @ 11 threads | U 45.04 D 7.78 kB/s | TPS 20 | Load 14.87%]0;PocketMine-MP 1.7dev-561 | Online 12/50 | Memory 249.51/3.95/429.34/1281.16 MB @ 11 threads | U 36.37 D 8.47 kB/s | TPS 19.68 | Load 18.56%]0;PocketMine-MP 1.7dev-561 | Online 12/50 | Memory 249.56/3.95/429.34/1281.16 MB @ 11 threads | U 42.24 D 8.18 kB/s | TPS 20 | Load 17.06%[36m[19:32:23] [m[37m[Server thread/INFO]: [32m[19:32:23] [32mGamerCube19[33m s'est déplacé du monde [31mlobby[33m vers le monde [32mminage[m
]0;PocketMine-MP 1.7dev-561 | Online 12/50 | Memory 251.85/3.95/430.38/1281.16 MB @ 11 threads | U 35.41 D 8.99 kB/s | TPS 20 | Load 18.7%bash : ligne 1 : 18982 Erreur de segmentation  ./start.sh ```
# Paste your crashdump here (if theres any)
```

```text
# Paste the contents of your TeaSpoonDump here ( by running /bugreport ) [can be found at <TeaSpoon Plugin Folder>/dumps/<time>]
```

**Comment by Antisa14200** [comment]
Have you tried using latest one from here? Even tho is has some issuses but if errors like those are not showing anymore, thats good

**Comment by NycuRO** [comment]
It's memory error and by software shit.

**Comment by Antisa14200** [comment]
Probably, as more players join

**Comment by teracube** [comment]
@Antisa14200 
yes i try the latest build Dev #89 =)  same issue
i try too a full rebuild of the server =) same issue

my server had memory-limit=2024M / 50 slots / 19 plugins / 6 worlds
no pb until today

i think it' a TeaSpoon issue or an new incompatibility with PMMP or another plugin

but no new information, no crashdump...

**Comment by Antisa14200** [comment]
Which version of pmmp you using?

**Comment by teracube** [comment]
at first crash 
PocketMine-MP version [33m1.7dev-554[m "[REDACTED]" (API 3.0.0-ALPHA10)
but same issue with latest build  PocketMine-MP_1.7dev-565_1e896eff_API-3.0.0-ALPHA10.phar
see attached file screenlog.zip

[screenlog.zip](https://github.com/CortexPE/TeaSpoon/files/1594774/screenlog.zip)

**Comment by Antisa14200** [comment]
Probably problem with what @NycuRO mentioned
Idk honestly, maybe @CortexPE knows more.

**Comment by teracube** [comment]
3 hours on time server without TeaSpoon  :-( and no problem
maybe memory error TeaSpoon / PMMP

**Comment by NycuRO** [comment]
try reboot .

**Comment by teracube** [comment]
@NycuRO 
I've already done this and no success
there is always the issue segmentation fault when TeaSpoon is enabble on my server.
memory conflict i suppose

**Comment by CortexPE** [comment]
wtf. what the f**k.
TeaSpoon doesnt modify / even use threading.
Idk how to reproduce this...
Its either PMMP's fault or PHP's... Can you post your server.log file before and after the seg fault error?

**Comment by teracube** [comment]
@CortexPE 
server.log file before
at the end : 
bash : ligne 1 : 24807 Erreur de segmentation  ./start.sh 
then server crash an no other log, no crashdump...
[screenlog.zip](https://github.com/CortexPE/TeaSpoon/files/1594989/screenlog.zip)

**Comment by teracube** [comment]
@CortexPE 
another file
only 1 player when server crash
teaspoon folder and nether+ender was deleted
server start with a new configuration of TeaSpoon then server crash
thanks

[issue.zip](https://github.com/CortexPE/TeaSpoon/files/1595037/issue.zip)

**Comment by teracube** [comment]
@CortexPE 

update

1/firt test :

server crash : segmentation fault .start.sh
If i increase RAM 2024=) 4048 server crash too

PocketMine-MP version 1.7dev-565 "[REDACTED]" (API 3.0.0-ALPHA10)
TeaSpoon_dev-89.phar
PiggyCustomEnchants_dev-180.phar

2/second test 

I remove PiggyCustomEnchants

PocketMine-MP version 1.7dev-565 "[REDACTED]" (API 3.0.0-ALPHA10)
TeaSpoon_dev-89.phar

server is ok since 2 hours (with TeaSpoon but without PCE)

in conclusion 

i suppose an incompatibility/conflict  with TeaSpoon ( dev > v81)  and (PCE dev >178)

**Comment by CortexPE** [comment]
try disabling every plugin except TeaSpoon

**Comment by CortexPE** [comment]
are you in a VPS or a MultiCraft host?

**Comment by teracube** [comment]
@CortexPE

ok i will try test server with TeaSpoon only but it need time because i do not have a second server test .

server is OVH hosted  with GSP-Panet not Multicraft .

**Comment by CortexPE** [comment]
```bash
#!/bin/bash
DIR="$(cd -P "$( dirname "${BASH_SOURCE[0]}" )" && pwd)"
cd "$DIR"

DO_LOOP="no"

while getopts "p:f:l" OPTION 2> /dev/null; do
	case ${OPTION} in
		p)
			PHP_BINARY="$OPTARG"
			;;
		f)
			POCKETMINE_FILE="$OPTARG"
			;;
		l)
			DO_LOOP="yes"
			;;
		\?)
			break
			;;
	esac
done

if [ "$PHP_BINARY" == "" ]; then
	if [ -f ./bin/php7/bin/php ]; then
		export PHPRC=""
		PHP_BINARY="./bin/php7/bin/php"
	elif [[ ! -z $(type php) ]]; then
		PHP_BINARY=$(type -p php)
	else
		echo "Couldn't find a working PHP 7 binary, please use the installer."
		exit 1
	fi
fi

if [ "$POCKETMINE_FILE" == "" ]; then
	if [ -f ./PocketMine-MP.phar ]; then
		POCKETMINE_FILE="-d xdebug.auto_trace=ON -d xdebug.trace_output_dir=SEGFAULTDEBUG/./PocketMine-MP.phar"
	elif [ -f ./src/pocketmine/PocketMine.php ]; then
		POCKETMINE_FILE="-d xdebug.auto_trace=ON -d xdebug.trace_output_dir=SEGFAULTDEBUG/ ./src/pocketmine/PocketMine.php"
	else
		echo "Couldn't find a valid PocketMine-MP installation"
		exit 1
	fi
fi

LOOPS=0

set +e
while [ "$LOOPS" -eq 0 ] || [ "$DO_LOOP" == "yes" ]; do
	if [ "$DO_LOOP" == "yes" ]; then
		"$PHP_BINARY" "$POCKETMINE_FILE" $@
	else
		exec "$PHP_BINARY" "$POCKETMINE_FILE" $@
	fi
	if [ "$DO_LOOP" == "yes" ]; then
		if [ ${LOOPS} -gt 0 ]; then
			echo "Restarted $LOOPS times"
		fi 
		echo "To escape the loop, press CTRL+C now. Otherwise, wait 5 seconds for the server to restart."
		echo ""
		sleep 5
		((LOOPS++))
	fi
done
```
try this wrapper (save it as <any file name>/sh) then do:
```
chmod +x <filename>.sh
./<filename>.sh
```

then let it segfault and send me the traces in the SEGFAULTDEBUG folder

**Comment by teracube** [comment]
@CortexPE 

Good News ; - )

no issue with latest build TeaSpoon and PMMP
- PocketMine-MP_1.7dev-570_a0a2ea01_API-3.0.0-ALPHA10.phar
- TeaSpoon_dev-90.phar

no issue/crash since 6 hours server online and 15 players average 

so no need to test wrapper
maybe an incompatibility between one version PMMP and one version TeaSpoon or high load memory 
but why i do not know

so i think i can close this issue for the moment

happy new year and thanks for your job

**Comment by Antisa14200** [comment]
Nice

Dana 1. sij 2018. 15:45 osoba "TeraCube" <notifications@github.com>
napisala je:

@CortexPE <https://github.com/cortexpe>

Good News ; - )

no issue with latest build TeaSpoon and PMMP

   - PocketMine-MP_1.7dev-570_a0a2ea01_API-3.0.0-ALPHA10.phar
   - TeaSpoon_dev-90.phar

no issue/crash since 6 hours server online and 15 players average

so no need to test wrapper
maybe an incompatibility between one version PMMP and one version TeaSpoon
or high load memory
but why i do not know

so i think i can close this issue for the moment

happy new year and thanks for your job

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/CortexPE/TeaSpoon/issues/136#issuecomment-354656883>,
or mute the thread

**Comment by teracube** [comment]
@CortexPE 
WTF issue reappear 10 hours after restarting server
I have a problem with the wrapper command
could you please check the attached file and the command

On the panel Rcon command i do :

chmod +x wrappersegfault.sh./wrappersegfault.sh

I have this error : Unknown command. Try /help for a list of commands

[wrappersegfault.zip](https://github.com/CortexPE/TeaSpoon/files/1596297/wrappersegfault.zip)

**Comment by NycuRO** [comment]
👎  seriosly teracube ? 👎

**Comment by NycuRO** [comment]
Why you created your server if you don't know how give one shit permission?

**Comment by NycuRO** [comment]
That's not a problem of teaspoon @CortexPE :/

**Comment by teracube** [comment]
@NycuRO 
sorry,server is hosted

**Comment by NycuRO** [comment]
@teracube after start, use chmod +x start.sh and before ./start.sh

**Comment by UnboltThread** [comment]
Where do I get TeaSpoon Plugin?

**Comment by CortexPE** [comment]
read the Installation section on the README
